drivers/sensor/wsen_pdus/wsen_pdus.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_WSEN_PDUS_WSEN_PDUS_H_
+#define ZEPHYR_DRIVERS_SENSOR_WSEN_PDUS_WSEN_PDUS_H_
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/sensor.h>
+
+#include <weplatform.h>
+
+#include "WSEN_PDUS_25131308XXX01.h"
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+#include <zephyr/drivers/i2c.h>
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c) */
+
+struct pdus_data {
+	/* WE sensor interface configuration */
+	WE_sensorInterface_t sensor_interface;
```

Comment:
You could infer this at build time and store in `struct pdus_config` instead

Reply:
Hello @MaureenHelm ,
the `sensor_interface` takes sensor "information" at runtime from the Wuerth Elektronik HAL sub module. Thus it may not be a "const" type (like `struct pdus_config`) .